src/main/java/Duke.java

```diff
@@ -1,103 +1,222 @@
-        takeUserInput();
+    private static String markCommandVerify(String details) {
+        if (details.matches("\\d+")){
+            int listNum = Integer.parseInt(details);
+            return (listLength >= listNum && listNum > 0) ? "isValid" : "isOutOfBounds";
+        }
+        return "isNotNum";
+    }
+
+    private static void invalidMarkArgs(String args, boolean isNum) {
+        if (isNum) {
+            System.out.print(LINE_PARTITION +
+                    String.format("\t  Task number %s does not exist.\n", args));
+        } else {
+            System.out.print(LINE_PARTITION +
+                    String.format("\t  %s is not a number.\n", args));
+        }
+        System.out.print(String.format("\t  Please use a number between 1 and %d for \"mark\" and \"unmark\" commands\n", listLength) +
+                LINE_PARTITION);
+    }
+    private static void noMarkArgs() {
+        System.out.print("You did not enter any number.\n" +
+                String.format("Please use a number between 1 and %d for \"mark\" commands\n", listLength) +
+                LINE_PARTITION);
+    }
+
+    private static void printMark(Task task) {
+        System.out.print(LINE_PARTITION +
+                String.format("\t  Great job! I will mark this task as done: \n\t  %s\n", task) +
+                LINE_PARTITION);
+    }
+
+    private static void printUnmark(Task task) {
+        System.out.print(LINE_PARTITION +
+                String.format("\t  Alright, I have marked this task as undone: \n\t  %s\n", task) +
+                LINE_PARTITION);
+    }
+    private static void executeTaskAdd(String args, String taskType) {
+        if (args == null) {
+            System.out.print(NO_ARGS_TASK_ERROR);
+        } else if (args.matches("(.*) /by (.*)") && taskType.equals("deadline")) {
+            deadlineAdd(args);
+        } else if (args.matches("(.*) /from (.*) /to (.*)") && taskType.equals("event")) {
+            eventAdd(args);
+        } else if (taskType.equals("todo") && !args.isEmpty()) {
+            todoAdd(args);
+        } else {
+            cmdFormatError();
+        }
+    }
+
+    private static void deadlineAdd(String args) {
+        String[] taskArgs = args.trim().split(" /by ", 2);
+        Deadline task = new Deadline(taskArgs[0], taskArgs[1]);
+        taskList[listLength] = task;
+        listLength++;
+        printTaskAdd();
+    }
+    private static void eventAdd(String args) {
+        String[] taskArgs = args.trim().split(" /from ", 2);
+        String[] taskDates = taskArgs[1].split(" /to ", 2);
+        Event task = new Event(taskArgs[0], taskDates[0], taskDates[1]);
+        taskList[listLength] = task;
+        listLength++;
+        printTaskAdd();
+    }
+    private static void todoAdd(String args) {
+        Todo task = new Todo(args);
+        taskList[listLength] = task;
+        listLength++;
+        printTaskAdd();
+    }
+    private static void cmdFormatError() {
```

Comment:
Maybe a verb can be used to increase clarity of function. I suggest printCmdError.